nodeup/pkg/model/networking/amazon-vpc-routed-eni.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networking
+
+import (
+	"k8s.io/kops/nodeup/pkg/model"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
+)
+
+// AmazonVPCRoutedENIBuilder writes the Amazon VPC CNI configuration
+type AmazonVPCRoutedENIBuilder struct {
+	*model.NodeupModelContext
+}
+
+var _ fi.NodeupModelBuilder = &AmazonVPCRoutedENIBuilder{}
+
+// Build is responsible for configuring the network cni
+func (b *AmazonVPCRoutedENIBuilder) Build(c *fi.NodeupModelBuilderContext) error {
+	if b.NodeupConfig.Networking.AmazonVPC == nil {
+		return nil
+	}
+
+	// Running Amazon VPC CNI on Ubuntu 22.04 and later requires setting MACAddressPolicy to `none` (ref: https://github.com/aws/amazon-vpc-cni-k8s/issues/2103 & https://github.com/kubernetes/kops/issues/16255)
+	if b.NodeupConfig.Networking.AmazonVPC != nil && b.Distribution.IsUbuntu() && b.Distribution.Version() >= 22.04 {
```

Comment:
```suggestion
	// Running on Ubuntu 22.04 and later requires setting MACAddressPolicy=none`
	// Ref: https://github.com/aws/amazon-vpc-cni-k8s/issues/2103 & https://github.com/kubernetes/kops/issues/16255
This should already be handled above
```suggestion
	if b.Distribution.IsUbuntu() && b.Distribution.Version() >= 22.04 {
```

Reply:
Ohhh right, good point. Thanks for catching that!